buildpacks/php/src/layers/composer_env.rs

```diff
@@ -0,0 +1,84 @@
+use crate::errors::PhpBuildpackError;
+use crate::PhpBuildpack;
+use libcnb::build::BuildContext;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::generic::GenericMetadata;
+use libcnb::layer::{ExistingLayerStrategy, Layer, LayerData, LayerResult, LayerResultBuilder};
+use libcnb::layer_env::{LayerEnv, ModificationBehavior, Scope};
+use libcnb::{Buildpack, Env};
+use libherokubuildpack::log::log_header;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+pub(crate) struct ComposerEnvLayer<'a> {
+    pub command_env: &'a Env,
+    pub dir: &'a PathBuf,
+}
+
+impl Layer for ComposerEnvLayer<'_> {
+    type Buildpack = PhpBuildpack;
+    type Metadata = GenericMetadata;
+
+    fn types(&self) -> LayerTypes {
+        LayerTypes {
+            build: true,
+            cache: false,
+            launch: true,
+        }
+    }
+
+    fn create(
+        &self,
+        _context: &BuildContext<Self::Buildpack>,
+        _layer_path: &Path,
+    ) -> Result<LayerResult<Self::Metadata>, <Self::Buildpack as Buildpack>::Error> {
+        log_header("Preparing Composer environment");
```

Comment:
Yeah, so... different buildpacks have it in different places, and for e.g. the bootstrap layer, it has to announce that it's restoring from cache... but in other layers, I do not want to log the header in the layer, since I use it multiple times for different purposes (platform for platform and webservers install)